packages/rs-sdk/src/platform/list.rs

```diff
@@ -0,0 +1,107 @@
+//! List Module
+//!
+//! This module provides a trait to fetch a list of objects from the platform.
+//!
+//! ## Traits
+//! - `List`: An async trait that lists items of a specific type from the platform.
+
+use dapi_grpc::platform::v0::GetDocumentsResponse;
+use dpp::document::Document;
+use drive_proof_verifier::proof::from_proof::FromProof;
+use rs_dapi_client::{transport::TransportRequest, DapiRequest, RequestSettings};
+
+use crate::{
+    error::Error,
+    mock::MockResponse,
+    platform::{document_query::DocumentQuery, query::Query},
+    Sdk,
+};
+
+/// Trait implemented by objects that can be listed or searched.
+#[async_trait::async_trait]
+pub trait List
```

Comment:
Rename to FetchList or FetchMany (I think FetchMany is better but would be okay with FetchList)